### DISCUSSION TITLE
flaiR (R Package) Discussions!

### DISCUSSION THREAD

**Discussion by davidycliao** [davidycliao/flaiR#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by joufeihuang** [comment]
Hello, I experienced this problem in installing this package. I am not sure how to resolve it. 
`> library(flaiR)
 flaiR: An R Wrapper for Accessing Flair NLP Tagging Features      
 Python : ✗                                     
 Flair: ✗                                     
 Flair is installing from Python                         
Error: 套件或命名空間載入失敗 ‘flaiR’:
 attachNamespace() 裡算 'flaiR' 時 .onAttach 失敗了，詳細內容：
  呼叫: stop_no_virtualenv_starter(version)
  錯誤: Suitable Python installation for creating a venv not found.
Please install Python with one of following methods:
- https://www.python.org/downloads/
- reticulate::install_python(version = '<version>')
In addition: Warning message:
R graphics engine version 16 is not supported by this version of RStudio. The Plots tab will be disabled until a newer version of RStudio is installed. `

**Comment by davidycliao** [comment]
Thank you for your  @joufeihuang .
I believe you've successfully downloaded it. From what I can tell, you've installed Python. However, I suspect your Python version might need to be 3.7 or higher. Otherwise, the Python Flair NLP may not automatically integrate when you load the flaiR package in R.

There are a few steps you can take:

1. update Python to 3.7. Then, you can directly use the reticulate package in R to install the Flair Python library. Use the following commands:

```

library(reticulate)
# This will tell you where Python is installed. py_config() 
# Activate the Python environment.
reticulate::use_python(reticulate::py_config()$python) 
# Install the Flair library in the specified Python environment.
reticulate::py_install("flair") 

```

After that, reload the flaiR package in R again. 
Hope it works

**Reply by joufeihuang** [reply]
Thank you for your reply! Great work!

**Reply by davidycliao** [reply]
Please feel free to let me know if this does not work.

**Comment by paula-montano** [comment]
Hi, after re-installing Python on my Windows PC, the flaiR installation via GitHub repo went straightforward. No issues at all! 
When re-installing Python, make sure to add Python to the path. 
Cool work!
